rasa/core/migrator.py

```diff
@@ -0,0 +1,214 @@
+import itertools
+import logging
+from typing import Text, Optional, List, Set, Dict, Any
+
+from tqdm import tqdm
+
+import rasa.cli.utils as cli_utils
+from rasa.core.brokers.broker import EventBroker
+from rasa.core.tracker_store import TrackerStore
+from rasa.core.trackers import EventVerbosity
+from rasa.exceptions import (
+    NoEventsToMigrate,
+    NoConversationsInTrackerStore,
+    NoEventsInTimeRange,
+    PublishingError,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Migrator:
+    """Manages the publishing of events in a tracker store to an event broker."""
+
+    def __init__(
+        self,
+        tracker_store: TrackerStore,
+        event_broker: EventBroker,
+        endpoints_path: Text,
+        requested_conversation_ids: Optional[Text] = None,
+        minimum_timestamp: Optional[float] = None,
+        maximum_timestamp: Optional[float] = None,
+    ):
+        """
+        Args:
+            endpoints_path: Path to the endpoints file used to configure the event
+                broker and tracker store. If `None`, the default path ('endpoints.yml')
+                is used.
+            requested_conversation_ids: List of conversation IDs requested to be
+                processed.
+            minimum_timestamp: Minimum timestamp of events that are published.
+                If `None`, apply no such constraint.
+            maximum_timestamp: Maximum timestamp of events that are published.
+                If `None`, apply no such constraint.
+
+        """
+        self.endpoints_path = endpoints_path
+        self.tracker_store = tracker_store
+        self.event_broker = event_broker
+        self.requested_conversation_ids = requested_conversation_ids
+        self.minimum_timestamp = minimum_timestamp
+        self.maximum_timestamp = maximum_timestamp
+
+    def publish_events(self) -> int:
+        """Publish events in a tracker store using an event broker.
+
+        Exits if the publishing of events is interrupted due to an error. In that case,
```

Comment:
random thought: should we log the max timestamp we found so that users know until which point things were migrated?

Reply:
that's known from the `minimum_timestamp` that's printed in the continuation message